@jaypie/dynamodb@0.6.4 imports ConflictError (errors ≥1.2.3) but ecosystem pins @jaypie/errors@1.2.1 → unbundleable

**Description**

`@jaypie/dynamodb@0.6.4` imports `ConflictError` from `@jaypie/errors`, but `ConflictError` was only added in `@jaypie/errors@1.2.3`. `@jaypie/dynamodb` declares `@jaypie/errors` only as a wildcard **peer** (`"*"`), so it does not force `>=1.2.3`. The current `jaypie@1.2.47` meta-package and its siblings (`@jaypie/core@1.2.2`, `@jaypie/aws`, `@jaypie/express`, `@jaypie/lambda`, `@jaypie/constructs`) all resolve `@jaypie/errors` to `1.2.1`, which does **not** export `ConflictError`. On a standard install the `*` peer is satisfied by the already-deduped `1.2.1`, and any bundler (esbuild) then fails to resolve the import. This is a version skew *inside* the Jaypie ecosystem rather than a consumer mistake.

**Context**

- Package: `@jaypie/dynamodb@0.6.4`
- Related: `@jaypie/errors` (`1.2.1` vs `1.2.3`), `jaypie@1.2.47`
- Environment: npm-workspaces monorepo, esbuild Lambda bundling (Node 24, ESM, `@aws-sdk/*` + node builtins external), reproduces on macOS and CI Linux

Evidence from the published `@jaypie/dynamodb@0.6.4`:
- `dist/esm/index.js:3` → `import { ConfigurationError, ConflictError } from '@jaypie/errors';`
- `dist/cjs/index.cjs:370` → `throw new errors.ConflictError("A conditional write in the transaction was rejected because the item already exists");`
- `package.json` `peerDependencies`: `"@jaypie/errors": "*"` (no floor)

`@jaypie/errors`:
- `1.2.1`: does **not** export `ConflictError`
- `1.2.3`: exports `ConflictError`

**Steps to Reproduce**

1. In a project that depends on `jaypie@1.2.47` (which transitively pins `@jaypie/errors@1.2.1`), add `@jaypie/dynamodb@^0.6.1` and let it resolve to `0.6.4`.
2. Bundle a module that imports from `@jaypie/dynamodb` with esbuild (`format: 'esm'`, `target: 'node24'`, `@aws-sdk/*` + node builtins marked external).
3. The build fails:
   ```
   ✘ [ERROR] No matching export in "node_modules/@jaypie/errors/dist/esm/index.js" for import "ConflictError"
       3 │ import { ConfigurationError, ConflictError } from '@jaypie/errors';
   ```
   (The CJS/runtime equivalent surfaces as `errors.ConflictError is not a constructor`.)

**Expected Behavior**

Installing `@jaypie/dynamodb@0.6.4` alongside the current `jaypie` meta-package resolves a `@jaypie/errors` that exports `ConflictError`, and bundling succeeds.

**Actual Behavior**

`@jaypie/errors` dedupes to `1.2.1` (no `ConflictError`), so esbuild cannot resolve the import and the bundle fails. `@jaypie/dynamodb@0.6.4` is effectively unbundleable in any project on the current `jaypie@1.2.x` line until `@jaypie/errors` is independently bumped.

**Suggested Fix**

- Raise the floor of `@jaypie/dynamodb`'s `@jaypie/errors` peer (and/or runtime dep) to `>=1.2.3` — the version that introduced `ConflictError`.
- Bump `@jaypie/errors` to `>=1.2.3` in lockstep across the `jaypie@1.2.x` meta-package and siblings so a single deduped version satisfies everyone.

**Workaround**

Pin `@jaypie/dynamodb` to `0.6.1` (the last release that does not import `ConflictError`).

--- a/packages/dynamodb/package.json
+++ b/packages/dynamodb/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@jaypie/dynamodb",
-  "version": "0.6.4",
+  "version": "0.6.5",
   "repository": {
     "type": "git",
     "url": "git+https://github.com/finlaysonstudio/jaypie.git"
@@ -58,7 +58,7 @@
     "typescript": "^5.3.3"
   },
   "peerDependencies": {
-    "@jaypie/errors": "*",
+    "@jaypie/errors": "^1.2.3",
     "@jaypie/kit": "*",
     "@jaypie/logger": "*",
     "@modelcontextprotocol/sdk": ">=1.0.0"

--- a/packages/dynamodb/src/__tests__/packageDependencies.spec.ts
+++ b/packages/dynamodb/src/__tests__/packageDependencies.spec.ts
@@ -1,0 +1,48 @@
+import { readFileSync } from "node:fs";
+import { dirname, resolve } from "node:path";
+import { fileURLToPath } from "node:url";
+
+import { satisfies } from "semver";
+import { describe, expect, it } from "vitest";
+
+//
+//
+// Constants
+//
+
+// The version of @jaypie/errors that introduced ConflictError, which
+// entities.ts imports. Installs that resolve an older @jaypie/errors fail to
+// bundle ("No matching export ... for import ConflictError"). See issue #400.
+const CONFLICT_ERROR_FLOOR = "1.2.3";
+const PRE_CONFLICT_ERROR = "1.2.1";
+
+const packageJson = JSON.parse(
+  readFileSync(
+    resolve(dirname(fileURLToPath(import.meta.url)), "../../package.json"),
+    "utf8",
+  ),
+);
+
+//
+//
+// Run tests
+//
+
+describe("@jaypie/dynamodb dependency floors", () => {
+  it("imports ConflictError from @jaypie/errors", () => {
+    const entities = readFileSync(
+      resolve(dirname(fileURLToPath(import.meta.url)), "../entities.ts"),
+      "utf8",
+    );
+    expect(entities).toContain("ConflictError");
+    expect(entities).toContain('from "@jaypie/errors"');
+  });
+
+  it("declares an @jaypie/errors peer that requires ConflictError support", () => {
+    const range = packageJson.peerDependencies?.["@jaypie/errors"];
+    expect(range).toBeString();
+    // A floorless "*" satisfies the pre-ConflictError 1.2.1, which breaks bundling
+    expect(satisfies(PRE_CONFLICT_ERROR, range)).toBe(false);
+    expect(satisfies(CONFLICT_ERROR_FLOOR, range)).toBe(true);
+  });
+});

--- a/packages/mcp/package.json
+++ b/packages/mcp/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@jaypie/mcp",
-  "version": "0.8.85",
+  "version": "0.8.86",
   "description": "Jaypie MCP",
   "repository": {
     "type": "git",

--- a/packages/mcp/release-notes/dynamodb/0.6.5.md
+++ b/packages/mcp/release-notes/dynamodb/0.6.5.md
@@ -1,0 +1,18 @@
+---
+version: 0.6.5
+date: 2026-06-28
+summary: Raise @jaypie/errors peer floor to ^1.2.3 so ConflictError resolves
+---
+
+## Fixes
+
+- Raise the `@jaypie/errors` peer dependency from a floorless `*` to `^1.2.3`.
+- `0.6.4` imports `ConflictError`, which `@jaypie/errors` only began exporting in
+  `1.2.3`. The `*` peer was satisfied by an already-deduped older `@jaypie/errors`
+  (e.g. `1.2.1`), so esbuild and other bundlers failed with
+  `No matching export ... for import "ConflictError"` (and the runtime equivalent
+  `errors.ConflictError is not a constructor`).
+- With the floor in place, installs resolve an `@jaypie/errors` that exports
+  `ConflictError` and bundling succeeds.
+
+No runtime behavior changes.

--- a/packages/mcp/release-notes/mcp/0.8.86.md
+++ b/packages/mcp/release-notes/mcp/0.8.86.md
@@ -1,0 +1,9 @@
+---
+version: 0.8.86
+date: 2026-06-28
+summary: Ship @jaypie/dynamodb 0.6.5 release notes
+---
+
+## Changes
+
+- Bundle the `@jaypie/dynamodb` 0.6.5 release notes (errors peer floor fix).

